include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -242,19 +308,55 @@ inline typename seq_type::iterator erase_gap(seq_type & seq, typename seq_type::
  * the container. Before delegating, the function checks if the range
  * [\p first, \p last) contains only seqan3::gap symbols.
  */
-template <sequence_container_concept seq_type>
+template <sequence_container_concept aligned_seq_t>
 //!\cond
-    requires weakly_assignable_concept<reference_t<seq_type>, gap const &>
+    requires detail::is_gapped_alphabet<value_type_t<aligned_seq_t>>
 //!\endcond
-inline typename seq_type::iterator erase_gap(seq_type & seq,
-                                             typename seq_type::const_iterator first,
-                                             typename seq_type::const_iterator last)
+inline typename aligned_seq_t::iterator erase_gap(aligned_seq_t & aligned_seq,
+                                                  typename aligned_seq_t::const_iterator first,
+                                                  typename aligned_seq_t::const_iterator last)
 {
     for (auto it = first; it != last; ++it)
         if (*it != gap{}) // [[unlikely]]
             throw gap_erase_failure("The range to be erased contains at least one non-gap character.");
 
-    return seq.erase(first, last);
+    return aligned_seq.erase(first, last);
+}
+
+/*!\brief An implementation of seqan3::aligned_sequence_concept::assign_unaligned_sequence for sequence containers.
+ * \tparam        aligned_seq_t     Type of the container to reassign; must model seqan3::sequence_container_concept;
+ *                                  the value type must be a seqan3::gapped alphabet.
+ * \tparam        unaligned_seq_t   Type of the container to assign from; must model std::ranges::ForwardRange;
+ * \param[in,out] aligned_seq       The gapped sequence container to assign to.
+ * \param[in,out] unaligned_seq     The unaligned sequence container to assign from.
+ *
+ * \relates seqan3::aligned_sequence_concept
+ *
+ * \details
+ *
+ * This function clears the content of the `gapped` container and reassigns the content of the `unaligned` container by
+ * using std::copy.
+ *
+ * ### Performance
+ *
+ * Linear in the size of unaligned_seq.
+ *
+ * ### Exceptions
+ *
+ * Strong exception guarantee.
+ *
+ */
+template <sequence_container_concept aligned_seq_t, std::ranges::ForwardRange unaligned_sequence_type>
+//!\cond
+    requires detail::is_gapped_alphabet<value_type_t<aligned_seq_t>> &&
+             weakly_assignable_concept<reference_t<aligned_seq_t>, reference_t<unaligned_sequence_type>>
+//!\endcond
+void assign_unaligned(aligned_seq_t & aligned_seq, unaligned_sequence_type && unaligned_seq)
+{
+    aligned_seq_t tmp;
+    tmp.resize(std::ranges::size(unaligned_seq));
```

Comment:
According to [this](http://eel.is/c++draft/range.prim.size) I really think that you can't use `size` without requiring a `SizedRange`, can you? 
You need to use std::ranges::distance(unaligned_seq), which will resolve to size if it can otherwise call distance(end, begin), while doing the most performant operation for this (see [here](http://eel.is/c++draft/range.iter.op.distance))